src/components/autocomplete/js/autocompleteController.js

```diff
@@ -327,9 +334,8 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
 
         // cancel results if search text is not long enough
         if (!isMinLengthMet()) {
-          ctrl.loading = false;
           ctrl.matches = [];
-          ctrl.hidden  = shouldHide();
+          setLoading(false);
```

Comment:
So, the reason I put it in a method was because the existing way caused it to (very often) wait until the next digest cycle before checking whether or not to be hidden. This isn't a huge deal, but it means that we sometimes show the suggestion list too early/too late. This is particularly apparent on IE which has slower performance due to the number of digest cycles.

Additionally, I just updated it to use you're suggested approach and it breaks the ability to have an `ng-click`able element inside the `md-not-found`. I'm sure I can dig in and figure it out if you think it's important to change this, but I feel like it's a good performance optimization.